asset/fetcher.go

```diff
@@ -124,21 +124,28 @@ func (h *httpFetcher) Fetch(ctx context.Context, url string, headers map[string]
 		return nil, fmt.Errorf("can't open tmp file for asset: %s", err)
 	}
 
+	cleanup := func() {
+		tmpFile.Close()
+		_ = os.Remove(tmpFile.Name())
+	}
+
```

Comment:
This is good, but a pattern that is more resistant to future regressions might be:

```
defer func() {
    if err != nil {
        tmpFile.Close()
        _ = os.Remove(tmpFile.Name())
    }
}()
```

Where err is a named return variable. Then the function can return from anywhere, without the risk of someone forgetting to call cleanup().